Epic 6919 : Update virus metadata in Learn

labkey-bpatel opened the pull request:
- Update AssayNAbAntigenMetadata tables & ETL script.
- Create AntigenPanelMetadata tables & ETL script.
- Create VirusPanel tables & ETL script.
- Update Antigen tab to include new columns, and minor styling changes.
- Feature Request 39428: Modify home page to clarify for HIV

labkey-sweta committed:
Teamcity test failure fix (#258)

labkey-bpatel committed:
Epic 6919: Update virus metadata in Learn:
- Update AssayNAbAntigenMetadata tables
- Create AntigenPanelMetadata & VirusPanel tables
- ETL updates
- Updates related to showing new metadata under Antigen tab for NAb and NAb MAb
- Add Sample Data

labkey-bpatel committed:
Styling changes for Panel values. Adjust sample data for panels.

labkey-bpatel committed:
Column width and flex adjustments. Additional sample data with long virus name.

labkey-bpatel committed:
Merge remote-tracking branch 'origin/release20.3-SNAPSHOT' into 20.3_fb_virus_metadata_6919

labkey-bpatel committed:
Remove overflowY flag to avoid having an external scrollbar (which results in two scroll bars).

labkey-bpatel committed:
Upgrade module version.

labkey-bpatel committed:
Feature Request 39428: Modify home page to clarify for HIV

labkey-bpatel committed:
Remove comma

labkey-bpatel committed:
Automated test - newly added antigen metadata cols for both NAb and NAb MAb antigen tabs; and update to home page header text.

labkey-bpatel committed:
Automated test fix.

labkey-bpatel committed:
Fix source query name.

labkey-bpatel committed:
verifyPKLinePlot test fix.

labkey-bpatel committed:
Fix CDSGridTest.verifyGrid() "Unable to locate label: Assay identifier" error by setting ShowHiddenVariables to true.

labkey-bpatel committed:
Set and reset ShowHiddenVariables for all tests in CDSGridTest (instead of per test).

labkey-bpatel committed:
Fix for Issue 39980: Learn/Assay/Nab/Antigen - scrolling and sizing issue

XingY commented on resources/queries/cds/nabAntigenWithPanelMeta.sql:
```diff
@@ -0,0 +1,40 @@
+ag.virus_host_cell,
+ag.virus_backbone,
+ag.cds_virus_id,
+GROUP_CONCAT(DISTINCT ap.panel_name, ';') AS panel_names
```
can panel_name itself contain ';'? if so, the ui can't split it correctly.

labkey-bpatel replied:
Hmm good point. I am using least likely one "|" instead - what do you think?
From handful of sample data client had mentioned, it looked like a pretty straight forward text.

XingY commented on resources/etls/LoadApplication.xml:
```diff
@@ -144,6 +144,14 @@
             <source schemaName="cds" queryName="ds_bamaantigen" />
             <destination schemaName="cds" queryName="BAMAAntigen" targetOption="truncate" />
         </transform>
+        <transform id="PopulateAntigenPanel" type="org.labkey.di.pipeline.TransformTask">
```
I know we are using this pattern of import table -> ds_ query -> target table a lot, but I believe the ds_antigenPanel and ds_virusPanel queries are not necessary since they are identical to their import_ counterparts

labkey-bpatel replied:
Now removed, and using the table names instead.

XingY commented on resources/schemas/dbscripts/postgresql/cds-20.001-20.002.sql:
```diff
@@ -0,0 +1,87 @@
+
+CREATE TABLE cds.import_virusPanel
+(
+    rowid SERIAL,
```
we haven't used rowid for cds. is there a reason to use them here? Why can't pk just be (container, cds_virus_id, cds_panel_id)? Since cds tables are repopulated each time etl is re-run, easiest is to not using identity column. I do see import_NABMAb table has a row_id, i believe the reason is part of the actual key contains a float data type which is not accepted PK so row_id is used as db pk as workaround

labkey-bpatel replied:
Ah got it! Now removed!

labkey-bpatel committed:
Code review changes:
- Change delimiter.
- Remove un-necessary ds_ queries & related update to etl xml.
- Remove rowid for viralPanel tables.

labkey-bpatel committed:
Revert Fix for Issue 39980: Learn/Assay/Nab/Antigen - scrolling and sizing issue

labkey-bpatel committed:
Remove outer scroll bar.

labkey-bpatel committed:
Fix for Issue 39980: scrolling and resizing issues

labkey-bpatel committed:
Use Ext.getCmp() instead of digging into child components.

labkey-bpatel commented:
Two unrelated & same test failures as the default branch. Merging!
